tox.ini

```diff
@@ -138,44 +143,93 @@ changedir = {[base]changedir}
 commands =
     coveralls
 
-[testenv:ensure_non_running_provider]
+# LSR_MOLECULE_DEPS may contain aditional Molecule dependencies. For example,
+# in `network` system role, LSR_MOLECULE_DEPS can be set as
+#
+#   LSR_MOLECULE_DEPS='-rmolecule_requirements.txt'
+#
+# where `molecule_requirements.txt` contains two lines:
+#
+#   jmespath
+#   selinux
+#
+[molecule]
 deps =
-    PyYAML
-changedir = {toxinidir}/tests
-commands = {toxinidir}/tests/ensure_non_running_provider.py
+    docker
+    molecule
+    {env:LSR_MOLECULE_DEPS:}
 
 [testenv:molecule_lint]
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
 commands_pre =
     molecule --version
     ansible --version
-commands = molecule {posargs} lint
+commands =
+    molecule {posargs} lint
 
 [testenv:molecule_syntax]
-commands = molecule {posargs} syntax
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
+commands =
+    molecule {posargs} syntax
 
 [testenv:molecule_test]
-commands = molecule {posargs} test
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
+commands =
+    molecule {posargs} test
```

Comment:
This and other molecule commands only allow for testing the default scenario by default. You can manually specify other scenarios by lightly abusing posargs (`tox -e molecule_test -- -s scenarioname`), but that won't help much in CI. 

molecule testing is heavily scoped in to the scenario, and having multiple scenarios per role is common. I recommend handling support for multiple scenarios now, before potentially using this is a template for other roles tested with molecule scenarios.

Reply:
@seandst any idea how to do that?

Reply:
Sadly, `tox` is not so expressive when one needs to implement something a bit more complex than substitution in it. My idea how to handle the support for multiple scenarios is make a python wrapper around Molecule.

Reply:
Why can't you just set the scenario name in environment variables straight from .travis.yml?

Reply:
Having something like `molecule -s {env:LSR_MSCENARIO:default} {posargs} test` in `tox` is of course possible, but if we want to run a test matrix where for every scenario from the role specific scenarios run a set of molecule commands, I am afraid that environment variables will be not enough (but maybe there exists a workaround in Molecule that I am not yet familiar with).